### PR TITLE
Add jooq

### LINKED ISSUE #15: Add JOOQ to your-rent-data
Add JOOQ and database management into the your-rent-data project.

- [x] Add test PostgreSQL database with testcontainers
- [x] Add flyway configuration for database generation, updating and versioning
- [x] Add JOOQ with code generation
- [x] Evaluate the current model in your-rent-core and possibly migrate it to your-rent-data

### PR DESCRIPTION
**Pull request by benfante**
Resolves #15

### CODE REVIEW

File: your-rents-data-jooq/.mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
+/*
```
Comment (egch) [inline your-rents-data-jooq/.mvn/wrapper/MavenWrapperDownloader.java:1]: Do we need to have this in git?

File: your-rents-data/src/test/java/com/yourrents/data/BaseDataTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (egch) [inline your-rents-data/src/test/java/com/yourrents/data/BaseDataTest.java:23]: my personal taste, I'd prefer to use assertJ instead

**Review by egch — APPROVED** [review]
great work! ✔️ 
I am just not sure if we really need to add the _.mvn_ folder with all its contents to git.

File: your-rents-data-jooq/.mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (benfante) [reply your-rents-data-jooq/.mvn/wrapper/MavenWrapperDownloader.java:1]: Yes, it allows to auto-download Maven if it's not already present.

File: your-rents-data/src/test/java/com/yourrents/data/BaseDataTest.java (hunk 2 above)
Comment (benfante) [reply your-rents-data/src/test/java/com/yourrents/data/BaseDataTest.java:23]: We can use both, or more. My preference is for Hamcrest.

File: your-rents-data-jooq/.mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (benfante) [reply your-rents-data-jooq/.mvn/wrapper/MavenWrapperDownloader.java:1]: In fact we can remove the maven wrapper jar, but not the MavenWrapperDownloader.java. I will remove it from all projects. https://github.com/takari/maven-wrapper#usage-without-binary-jar

File: your-rents-data-jooq/src/test/java/com/yourrents/data/jooq/util/ExportDevDataToJson.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+import org.jooq.DSLContext;
+import org.jooq.impl.DSL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExportDevDataToJson {
+    static final Logger log = LoggerFactory.getLogger(ExportDevDataToJson.class);
```
Comment (egch) [inline your-rents-data-jooq/src/test/java/com/yourrents/data/jooq/util/ExportDevDataToJson.java:37]: Maybe Lombok for the log?

Comment (benfante) [reply your-rents-data-jooq/src/test/java/com/yourrents/data/jooq/util/ExportDevDataToJson.java:37]: In that module there is no other need of Lombok. I wouldn't introduce it just for the log.

**Pull request merged**